### PR TITLE
Support Config.node false value

### LINKED ISSUE #209: `Config.node` is incorrectly assumed to always be an object
webpack-chain [makes the incorrect assumption](https://github.com/neutrinojs/webpack-chain/blob/c59f853e7164f0acac9970c230a35500a1e5cbd3/src/Config.js#L122) that `Config.node` can never be anything other than an object, despite [webpack allowing it to be set to `false`](https://github.com/webpack/webpack.js.org/blob/f799227e966cc40aa90a1c0df975208608326f9b/src/content/configuration/node.md#node).

As a result, calling `config.set('node', false)` results in the value being lost and the `node` property not existing in the object generated with `Config.toConfig()`.

### PR DESCRIPTION
**Pull request by SuperOleg39**
Resolves #209.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -842,6 +842,13 @@ config.resolve
 
 #### Config node
 
+You can set `node` option to object, or a `false` value.
+`false` value always rewrite `config.node` object.
```
Comment (opl-) [inline README.md:846]:
```suggestion
You can set the `node` option to an object, or a `false` value.
A `false` value always overrides the `config.node` chained map.
```

File: README.md [hunk 2]
```diff
@@ -842,6 +842,13 @@ config.resolve
...
+
+```js
+config.set('node', false);
```
Comment (opl-) [inline README.md:849]:
```suggestion
config.set('node', false);

// To use the value generated from `config.node` ChainedMap again
config.delete('node');
```

File: src/Config.js [hunk 3]
```diff
@@ -165,12 +166,21 @@ module.exports = class extends ChainedMap {
       );
     }
 
+    if (!omit.includes('node') && 'node' in obj) {
+      if (typeof obj.node === 'object') {
```
Comment (opl-) [inline src/Config.js:170]: `true` for `obj.node == []`. Though the input is unexpected it might cause issues.

**Review by opl- — CHANGES_REQUESTED** [review]
One problem I have with this approach is that doing `config.set('node', {})` will force the node property to never show up in the final config, which might end up being extremely confusing to a user who might not expect `config.delete('node')` to be the way to bring back the standard behavior. Similarly, using `config.set('node', [])` will cause even more weirdness to happen.

I see three potential solutions to this problem:
- Ignore this and call it a feature,
- Ignore any value other than `false`,
- Override the `config.set` function and raise an error.

I'm not sure what the standard is in this library, so I'm just mentioning it and leaving any decisions up to the maintainers. If the first option is chosen, I'd recommend at least changing the documentation accordingly.

**Comment by opl-** [comment]
I'd also like to mention the possibility of creating [callable classes](https://stackoverflow.com/questions/36871299/how-to-extend-function-with-es6-classes) even though I feel it's rather hacky. Something like this could also be used for #222 (though that one could be resolved without weird hacks by introducing a breaking change and bumping the major).

**Comment by edmorley** [comment]
@SuperOleg39 thank you for the PR and @opl- thank you for the really helpful review/thoughts. I don't use webpack day to day any more, so am reliant on suggestions as to what direction it should take longer term. I agree with @opl- 's thoughts about potentially confusing UX. What do you both think as the best option out of those that are proposed above?

**Comment by edmorley** [comment]
Closing since this project is no longer maintained:
https://github.com/neutrinojs/webpack-chain/issues/358

**Pull request closed without merging**